storage/innobase/fil/fil0fil.cc

```diff
@@ -3836,9 +3836,31 @@ fil_space_t *fil_space_acquire_for_io(ulint space_id) {
   return (space);
 }
 
+/** Acquire a tablespace for reading or writing a block,
+when it could be dropped concurrently.
+@param[in]	id	tablespace ID
+@return	the tablespace
+@retval	NULL if missing */
+fil_space_t *fil_space_acquire_for_io_with_load(ulint space_id) {
```

Comment:
This variant should be unnecessary. the fil_space_get_flags() + fil_space_get() should work fine.

Reply:
"trunk"

Reply:
because tablespace cannot be dropped concurrently during redo ?